hathor/cli/run_node.py

```diff
@@ -324,6 +326,11 @@ def __init__(self, *, argv=None):
         elif args.testnet:
             os.environ['HATHOR_CONFIG_YAML'] = TESTNET_SETTINGS_FILEPATH
 
+        try:
+            HathorSettings()
+        except ValidationError as e:
+            raise PreInitializationError(e)
```

Comment:
An alternative is doing `raise PreInitializationError from e`, which still makes `e` show in the traceback as what caused PreInitializationError, this is called [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining).

Reply:
Cool, I didn't know about this, thanks. Changed in https://github.com/HathorNetwork/hathor-core/pull/611/commits/44615a281d8be68479e90e818e774ca1887c2a83. I also added `TypeError` in the `except` block, as it's raised when an extra, invalid attribute is set in `HathorSettings`.